server/forge/github/github.go

```diff
@@ -610,13 +604,21 @@ func (c *client) Hook(ctx context.Context, r *http.Request) (*model.Repo, *model
 		return nil, nil, err
 	}
 
-	if pipeline != nil && pipeline.Event == model.EventRelease && pipeline.Commit == "" {
-		tagName := strings.Split(pipeline.Ref, "/")[2]
-		sha, err := c.getTagCommitSHA(ctx, repo, tagName)
-		if err != nil {
-			return nil, nil, err
+	if pipeline != nil {
+		if pipeline.Event == model.EventRelease && pipeline.Commit.SHA == "" {
```

Comment:
Maybe a check if message is empty but sha is set to load the commit details would be cleaner and could help loading the corresponding data always if possible. Cculd be especially helpful when adding new events later on.